shadows/framework/src/main/java/org/robolectric/shadows/ShadowUIModeManager.java

```diff
@@ -0,0 +1,364 @@
+package org.robolectric.shadows;
+
+import static android.os.Build.VERSION_CODES.TIRAMISU;
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.robolectric.util.reflector.Reflector.reflector;
+
+import android.annotation.SystemApi;
+import android.app.IUiModeManager;
+import android.app.UiModeManager;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.content.res.Configuration;
+import android.os.Build.VERSION;
+import android.os.Build.VERSION_CODES;
+import android.os.IBinder;
+import android.provider.Settings;
+import com.android.internal.annotations.GuardedBy;
+import com.google.common.collect.ImmutableSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.HiddenApi;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+import org.robolectric.annotation.Resetter;
+import org.robolectric.util.reflector.Accessor;
+import org.robolectric.util.reflector.Constructor;
+import org.robolectric.util.reflector.ForType;
+import org.robolectric.util.reflector.Static;
+import org.robolectric.util.reflector.WithType;
+import org.robolectric.versioning.AndroidVersions.V;
+
+/**
+ * Shadow for {@link UiModeManager}.
+ *
+ * @deprecated This class is misspelled. Update references to use {@link ShadowUiModeManager}.
+ */
+@Implements(UiModeManager.class)
+@Deprecated
+public class ShadowUIModeManager extends ShadowUiModeManager {
```

Comment:
Shouldn't it be the other way around? The new class that extends from the deprecated one?
If someone uses `ShadowUiModeManager` (the new class), it won't work, since there's no logic there, right?

Reply:
> If someone uses ShadowUiModeManager (the new class), it won't work, since there's no logic there, right?

It works fine as far as I can tell. The tests all pass, and the tests use the new ShadowUiModeManager.

If we did it the other way around, and have ShadowUiModeManager extend ShadowUIModeManager, calling `shadowOf(UIModeManager )` will continue to return a ShadowUIModeManager object, and this cannot be cast to a ShadowUiModeManager.

Reply:
Ugh when I tried to invert it, there are compile errors on Windows and Mac, probably due to being case sensitive.